### PR TITLE
Feature/11 - Add support for namespaces

### PR DESCRIPTION
**Pull request by csm10495**
Add namespace support. Also swap UINT_32 to size_t where we are doing memory allocation so 64-bit can allocate more if desired.

### CODE REVIEW

**Comment by csm10495** [comment]
Made new push to try to get Linux to compile again.

File: cNVMe/Controller.cpp [hunk 1]
```diff
@@ -357,8 +360,17 @@ namespace cnvme
...
+					auto namespaceSelected = this->NamespaceIdToNamespace.find(command.NSID);
+					if (namespaceSelected != this->NamespaceIdToNamespace.end())
+					{
+						LOG_INFO("Grabbing Identify Namespace for NSID " + std::to_string(namespaceSelected->first));
+						auto identifyNamespaceStructure = namespaceSelected->second.getIdentifyNamespaceStructure();
+						memcpy_s(transferPayload.getBuffer(), transferPayload.getSize(), &identifyNamespaceStructure, sizeof(identifyNamespaceStructure));
+					}
+					else
+					{
+						memset(transferPayload.getBuffer(), 0, transferPayload.getSize()); // Technically, this is ok if we don't have a namespace with the given NSID
```
Comment (mfamis) [inline cNVMe/Controller.cpp:372]: Add comment in this region describing what causes you to get in this else block and where in the spec it states that you populate the output payload with zeroes.

File: cNVMe/Namespace.cpp [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+		{
+			// nothing to do yet.
+		}
+
+		identify::structures::IDENTIFY_NAMESPACE& Namespace::getIdentifyNamespaceStructure()
+		{
+			this->IdentifyNamespace.NLBAF = DEFAULT_NUMBER_OF_LBA_FORMAT; // support 512/4096/8192 byte sectors
+			this->IdentifyNamespace.LBAF[0].LBADS = 9; // 2^9 = 512
+			this->IdentifyNamespace.LBAF[1].LBADS = 12; // 2^12 = 4096
+			this->IdentifyNamespace.LBAF[2].LBADS = 13; // 2^13 = 8192
```
Comment (mfamis) [inline cNVMe/Namespace.cpp]: I feel like these should be constants (e.g. LBADS_512B, LBADS_4K, LBADS_8K)

File: cNVMe/Namespace.cpp [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+			auto currentLbaFormat = this->IdentifyNamespace.LBAF[this->IdentifyNamespace.FLBAS.CurrentLBAFormat];
+			ASSERT_IF(currentLbaFormat.LBADS < 9, "Minimum posssible selected LBADS should be 9. 2^9 = 512.");
```
Comment (mfamis) [inline cNVMe/Namespace.cpp]: How is it possible to hit this assert? Is this just for development?

**Review by mfamis — APPROVED** [review]

File: cNVMe/Namespace.cpp (hunk 2 above)
Comment (csm10495) [reply cNVMe/Namespace.cpp]: using a macro instead.

File: cNVMe/Controller.cpp (hunk 1 above)
Comment (csm10495) [reply cNVMe/Controller.cpp:372]: Figure 106 of NVMe 1.3. Added a footnote.

File: cNVMe/Namespace.cpp (hunk 3 above)
Comment (csm10495) [reply cNVMe/Namespace.cpp]: It really isn't just for dev/if something goes very wrong.

**Pull request merged**